### ISSUE TITLE
Eliminate nesting in SecretProviderClass

### ISSUE THREAD

**Issue by tam7t** [GoogleCloudPlatform/secrets-store-csi-driver-provider-gcp#6]
### TL;DR
The current CDR format has multiple levels of string unmarshalling:

```
apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
kind: SecretProviderClass
metadata:
  name: app-secrets
spec:
  provider: gcp
  parameters:
    secrets: |
      array:
        - |
          resourceName: "projects/$PROJECT_ID/secrets/test/versions/latest"
          fileName: "good1.txt"
        - |
          resourceName: "projects/$PROJECT_ID/secrets/test/versions/latest"
          fileName: "good2.txt"
```

The values of all fields under `parameters` must be strings based on the [CRD definition](https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/34a978c51db6b048873167c9e2003c145d2f5f23/deploy/secrets-store.csi.x-k8s.io_secretproviderclasses.yaml#L38-L42).

However the additional `array` field under `secrets` seems unnecessary although other providers follow this pattern ([vault](https://github.com/hashicorp/secrets-store-csi-driver-provider-vault/blob/0d0f48d95c66797aa5dba704888d1c98f4f44f70/examples/v1alpha1_secretproviderclass.yaml#L11-L16) and [azure](https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/master/examples/v1alpha1_secretproviderclass_podid.yaml#L11-L20))

### Design

**Proposal**

Move to 1 level of unmarshalling by removing the `array` by making the `secrets` field:

```
type Secret struct {
...
}
...

var secrets []Secret
yaml.Unmarshal([]byte(attrib["secrets"]), &secrets)
```

and the resulting yaml:

```
apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
kind: SecretProviderClass
metadata:
  name: app-secrets
spec:
  provider: gcp
  parameters:
    secrets: |
        - resourceName: "projects/$PROJECT_ID/secrets/test/versions/latest"
          fileName: "good1.txt"
        - resourceName: "projects/$PROJECT_ID/secrets/test/versions/latest"
          fileName: "good2.txt"
```

**Alternatives considered**

**Resources**

**Additional information**
